### PR TITLE
refactor: move tests to `fuels`

### PR DESCRIPTION
**Pull request by Salka1988**

### CODE REVIEW

**Comment by Salka1988** [comment]
Ill put change this PR to draft cause it need new fuel-core release to be functional. We'll need a fuel-core patch release with the latest hotfix from fuel-vm.

File: packages/fuels-abigen-macro/tests/test_projects/.gitignore [hunk 1]
```diff
@@ -1,4 +0,0 @@
-**/out
-**/target
-**/Forc.lock
-**/.gitignore
```
Comment (digorithm) [inline packages/fuels-abigen-macro/tests/test_projects/.gitignore:4]: I don't think this file should be deleted -- isn't it what prevents us from accidentally committing auto-generated files from `forc build`?

File: packages/fuels/Cargo.toml [hunk 2]
```diff
@@ -19,5 +19,17 @@ fuels-signers = { version = "0.17.0", path = "../fuels-signers" }
 fuels-test-helpers = { version = "0.17.0", path = "../fuels-test-helpers" }
 fuels-types = { version = "0.17.0", path = "../fuels-types" }
 
+[dev-dependencies]
+anyhow = "1.0.58"
+ctor = " 0.1"
+fuel-core = { version = "0.9", default-features = false }
+fuel-gql-client = { version = "0.9", default-features = false }
+fuels = { path = "../fuels", default-features = false }
```
Comment (digorithm) [inline packages/fuels/Cargo.toml:27]: I don't think we need to import `fuels` here anymore, as this is inside `fuels/`.

```suggestion
```

**Review by digorithm — CHANGES_REQUESTED** [review]

**Comment by digorithm** [comment]
Oh, one more thing: the branch is outdated!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -144,6 +145,14 @@ jobs:
...
           cargo test --target wasm32-unknown-unknown --all-targets --no-default-features
 
+      - name: Install Fuel Core
+        if: ${{ matrix.command == 'test' }}
+        run: |
+          curl -sSLf https://github.com/FuelLabs/fuel-core/releases/download/v${{ env.FUEL_CORE_VERSION }}/fuel-core-${{ env.FUEL_CORE_VERSION }}-x86_64-unknown-linux-gnu.tar.gz -L -o fuel-core.tar.gz
+          tar -xvf fuel-core.tar.gz          
+          chmod +x fuel-core-${{ env.FUEL_CORE_VERSION }}-x86_64-unknown-linux-gnu/fuel-core
+          mv fuel-core-${{ env.FUEL_CORE_VERSION }}-x86_64-unknown-linux-gnu/fuel-core /usr/local/bin/fuel-core
+
```
Comment (digorithm) [inline .github/workflows/ci.yml:155]: What's the current situation with regards to this? Are we now effectively _using_ `fuel-core` as a binary and running our tests against it?

Comment (Salka1988) [reply .github/workflows/ci.yml:155]: Yes! In this way, we download the defined release of fuel-core bin. from `https://github.com/FuelLabs/fuel-core/releases` and then we use it to run the tests.

**Comment by emilyaherbert** [comment]
Will this be able to go in today? (CC @vnepveu @digorithm)

**Comment by digorithm** [comment]
> Will this be able to go in today? (CC @vnepveu @digorithm)

Yup, I don't see why not! Just waiting for the build to pass and then it LGTM 😄

**Review by digorithm — APPROVED** [review]
LGTM!

**Pull request merged**